pipenv/patched/notpip/_internal/baseparser.py

```diff
@@ -138,7 +138,7 @@ def __init__(self, *args, **kwargs):
         isolated = kwargs.pop("isolated", False)
         self.config = Configuration(isolated)
 
-        assert self.name
+        assert self.name, 'Could not find Pipfile.'
```

Comment:
So this is probably not the best place to handle this, we really don't want to be making additional patches in this folder if possible -- note that this is patching a vendored copy of pip directly which is definitely not preferred.  Ideally we would handle this in pipenv's main codebase instead

Reply:
Agree, we should check the existence somewhere early in `do_install()` instead of patching the vendor.